lib/hammer_cli_foreman/compute_resource.rb

```diff
@@ -215,21 +251,44 @@ class AvailableStorageDomainsCommand < HammerCLIForeman::ListCommand
       action :available_storage_domains
       command_name 'storage-domains'
 
+      option '--cluster-name', "Name of cluster", _("Cluster name to search by"),
+        attribute_name: :option_cluster_id
+
+      option '--cluster-id', "ID of cluster", _("Deprecated, use cluster-name")
+
       output do
         field :id, _('Id')
         field :name, _('Name')
       end
 
+      def request_params
+        params = super
+        params['cluster_id'] = params['cluster_id'].gsub('/', '%2F')
+        params
+      end
+
       build_options
     end
 
     class AvailableStoragePodsCommand < HammerCLIForeman::ListCommand
       action :available_storage_pods
       command_name 'storage-pods'
 
+      option '--cluster-name', "Name of cluster", _("Cluster name to search by"),
+        attribute_name: :option_cluster_id
+
+      option '--cluster-id', "ID of cluster", _("Deprecated, use cluster-name")
```

Comment:
Those changes are quite repetitive. Let's extract them into a command extension:
```ruby
module HammerCLIForeman
  module CommandExtensions
    class ComputeResourceSubcommand < HammerCLI::CommandExtensions
      option '--cluster-id', 'ID', _('Cluster ID'),
             deprecated: _('Use --cluster-name instead')
      option '--cluster-name', 'NAME', _('Cluster name or path to search by'),
             attribute_name: :option_cluster_id

      request_params do |params|
        params['cluster_id'] = params['cluster_id'].gsub('/', '%2F') if params['cluster_id']
      end
    end
  end
end
```

And then we can update the related commands like:
```ruby
build_options without: :cluster_id

extend_with(
  HammerCLIForeman::CommandExtensions::ComputeResourceSubcommand.new(only: %i[option request_params])
)
```